torchdynamo/mutation_guard.py

```diff
@@ -83,14 +83,23 @@ def install_generation_tagging_init():
     Monkey patch torch.nn.Module.__init__ and torch.nn.Module.__setstate__
     so we can detect nn.Module instances created dynamically inside forward methods.
     """
-
     if getattr(Module, "___needs_generation_tag_patch", True):
         init = Module.__init__
 
+        def patched_new(cls, *args, **kwargs):
```

Comment:
Patching new takes us back to the original problem of #305. Patching `__new__`  makes it slower.

IIUC, If we patch `__new__`, we don't need to patch `__init__`.

Reply:
Good question. 

I ran this on the same bench as https://github.com/pytorch/torchdynamo/pull/305 (same script, 10 runs, averaged)

Without patched new: 98.88617604838024
With patched new: 98.93933895271553
And with neither: 98.00966040366337

So I think this is within an acceptable margin of slowdown, thoughts?

Reply:
Thanks, with eval batch down to 16, I get:

Without patched new: 97.34619600294747 
With patched new: 97.60542088329098
And with neither: 96.98565560222735 

Thoughts?

Reply:
There was a PR which increased the batch size of AlexNet which hides the issue. I reverted this PR when doing my benchmarking. https://github.com/pytorch/benchmark/pull/919

Reply:
for my sanity can you try benchmarking and moving everything into `__new__` to verify that the slowdown still occurs?